src/tablestore.rs

```diff
@@ -224,6 +217,23 @@ impl TableStore {
                 _ => SlateDBError::from(e),
             })?;
 
+        if let Some(cache) = self.block_cache.as_ref() {
+            if write_cache {
```

Comment:
Nit: Inverting these two if's seems more natural to me, and I'm guessing a bool check is slightly cheaper than pattern matching + as_ref?